rocksdb/tests/test_db.py

```diff
@@ -489,6 +490,44 @@ def test_prefix_iteritems(self):
         ret = takewhile(lambda item: item[0].startswith(b'00002'), it)
         self.assertEqual(ref, dict(ret))
 
+class TestFixedPrefixExtractor(TestHelper):
+    def setUp(self):
+        TestHelper.setUp(self)
+        opts = rocksdb.Options(create_if_missing=True, prefix_extractor=4)
```

Comment:
Why is this 4, and not 5?

Reply:
I was actually testing two things (initially with a capped prefix transform and then with fixed prefix transform):
- prefix size = 4, key size = 4
- prefix size = 4, but key size for prefix matching is size > 4

The first case is straightforward, but in the second case when you provide prefix `b'00002'` to search for, it should give the key `b'00001'` too.  I wasn't getting the desired behavior until I realized (right now) that lines 518 & 528 need to be modified.